### ISSUE TITLE
False negatives for `Style/RedundantLineContinuation`

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12402]
## Describe the solution you'd like

**bad**

```ruby
let :user1 do
  create(:user, advertiser_type1, \
         advertiser: advertiser1, \
         user_type: user_type1)
end

let :user2 do
  build_stubbed(:user, advertiser_type2,
                advertiser: advertiser2, \
                user_type: user_type2)
end

def make_request
  patch api_v1_users_path(owner), \
        params: params, headers: headers
end

expect(page).to have_text \
  'Consequatur tenetur laboriosam necessitatibus. ' \
  'Qui architecto earum aliquid.'
```

**good**

```ruby
let :user1 do
  create(:user, advertiser_type1,
         advertiser: advertiser1,
         user_type: user_type1)
end

let :user2 do
  build_stubbed(:user, advertiser_type2,
                advertiser: advertiser2,
                user_type: user_type2)
end

def make_request
  patch api_v1_users_path(owner),
        params: params, headers: headers
end

expect(page).to have_text \
  'Consequatur tenetur laboriosam necessitatibus. '
  'Qui architecto earum aliquid.'
```

## Rubocop

```console
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop -V
1.57.2 (using Parser 3.2.2.4, rubocop-ast 1.29.0, running on ruby 2.7.8) [x86_64-linux]
  - rubocop-capybara 2.19.0
  - rubocop-factory_bot 2.24.0
  - rubocop-performance 1.19.1
  - rubocop-rails 2.22.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.25.0
  - rubocop-thread_safety 0.5.1
```
